notebooks/DiCE_getting_started_feasible.ipynb

```diff
@@ -0,0 +1,1072 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Dataset for training Black Box Model\n",
+    "train_data_vae= d.data_df.copy()\n",
+    "\n",
+    "#Creating list of encoded categorical and continuous feature indices\n",
+    "encoded_categorical_feature_indexes = d.get_data_params()[2]     \n",
+    "encoded_continuous_feature_indexes=[]\n",
+    "data_size= len(d.encoded_feature_names)\n",
+    "for i in range(data_size):\n",
+    "    valid=1\n",
+    "    for v in encoded_categorical_feature_indexes:\n",
+    "        if i in v:\n",
+    "            valid=0\n",
+    "    if valid:\n",
+    "        encoded_continuous_feature_indexes.append(i)            \n",
+    "encoded_start_cat = len(encoded_continuous_feature_indexes)\n",
+    "        \n",
+    "#One Hot Encoding for categorical features\n",
+    "encoded_data = d.one_hot_encode_data(train_data_vae)\n",
+    "\n",
+    "# The output/outcome variable position altered due to one_hot_encoding for categorical features: (Cont feat, Outcome, Cat feat) \n",
+    "# Need to rearrange columns such that outcome variable comes at the last\n",
+    "cols = list(encoded_data.columns)\n",
+    "cols = cols[:encoded_start_cat] + cols[encoded_start_cat+1:] + [cols[encoded_start_cat]]\n",
+    "encoded_data = encoded_data[cols]     \n",
+    "\n",
+    "#Normlization for conitnuous features\n",
+    "encoded_data= d.normalize_data(encoded_data)\n",
+    "print(encoded_data.columns)\n",
+    "dataset = encoded_data.to_numpy()\n",
+    "\n",
+    "#Train, Val, Test Splits\n",
+    "np.random.shuffle(dataset)\n",
+    "test_size= int(0.2*dataset.shape[0])\n",
+    "val_dataset= dataset[:test_size]\n",
+    "train_dataset= dataset[test_size:]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Training\n",
+    "for epoch in range(50):\n",
+    "    np.random.shuffle(train_dataset)\n",
+    "    train_batches= np.array_split( train_dataset, train_dataset.shape[0]//batch_size ,axis=0 )    \n",
+    "    print('Epoch: ', epoch)\n",
+    "    train_acc=0.0\n",
+    "    for i in range(len(train_batches)):    \n",
+    "        optimizer.zero_grad()\n",
+    "        train_x= torch.tensor( train_batches[i][:,:-1] ).float() \n",
+    "        train_y= torch.tensor( train_batches[i][:,-1], dtype=torch.int64 )\n",
+    "        \n",
+    "        out= m(train_x)\n",
+    "        train_acc += torch.sum( torch.argmax(out, axis=1) == train_y )\n",
+    "        \n",
+    "        # Cross Entropy Loss\n",
+    "        loss= crieterion(out, train_y)\n",
+    "        #L2 Regularization\n",
+    "        weight_norm = torch.tensor(0.)\n",
+    "        for w in m.ann_model.parameters():\n",
+    "            weight_norm += w.norm().pow(1)\n",
```

Comment:
does this do L1 or L2? I think you can directly call w.norm(2). Make sure the comments are correct.